### PR TITLE
Big Calendar agenda

### PR DESCRIPTION
**Pull request by khelif96**
Time To Completion: 10 hours

This was a pretty extensive refactor to the dashboard which involved integrating react-big-calendar
As part of this i needed to install moment.js to handle localization of  time and dates. (Which introduced alot of fun problems with testing).
react-big-calendar required some css tweaks to go with the overall theme of the app
react-big-calendar was also nice enough to handle grouping of  dates so we no longer need the `<EventSummaryCollection />` component

react-big-calendar is a dynamic component that changes ui's and selectively renders components in the agenda view based on if the date has passed or not. This is nice from a usability perspective but isnt ideal for testing where you would want some consistency so in order to get passed testing i mocked the moment.format function which react-big-calendar uses to format dates so that it always thinks its the same date. 

The next hurdle was parsing the userData from the api since it was in such an irregular form it did require some js tricks but everything should work.

image

image

### CODE REVIEW

File: website/src/App.jsx [hunk 1]
```diff
@@ -44,6 +44,7 @@ class App extends PureComponent {
             <Route path="/join/class" component={ProtectedJoinClass} />
             <Route path="/dashboard/:optional?" component={ProtectedDashboardRouter} />
             <Route component={NotFound} />
+
```
Comment (ayushyamitabh) [inline website/src/App.jsx]: empty line?

**Comment by khelif96** [comment]
Yeah irregular is the wrong word just difficult to parse

File: website/src/App.jsx (hunk 1 above)
Comment (khelif96) [reply website/src/App.jsx]: Forgot about that created a temp view for debugging purposes took it out but forgot to remove the line

**Comment by khelif96** [comment]
No new API request is made it relies on the API request being passed down from dashboardrouter

**Pull request merged**